### PR TITLE
pkp/pkp-lib#6057 Refactor submission files

### PR DESCRIPTION
**Pull request by NateWr**

### CODE REVIEW

**Comment by NateWr** [comment]
@asmecher here ihe OPS PR. It should be pretty straightforward, but another set of eyes would be appreciated.

File: classes/services/SubmissionFileService.inc.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+	/**
+	 * Allow authors to upload to galley file stages
+	 *
+	 * @param string $hookName
+	 * @param array $args [
+	 * ]
+	 */
+	public function modifyAssignedFileStages($hookName, $args) {
+		$allowedFileStages = &$args[0];
```
Comment (asmecher) [inline classes/services/SubmissionFileService.inc.php:65]: (Prefer `$a =& $b` to `$a = &$b` for readability/consistency.)

**Review by asmecher — CHANGES_REQUESTED** [review]
Looks good, Nate! Just a couple of comments.

File: classes/migration/upgrade/OPSv3_3_0UpgradeMigration.inc.php [hunk 2]
```diff
@@ -151,4 +152,18 @@ private function _isNumerical($array) {
...
+	 * Complete submission file migrations specific to OPS
+	 *
+	 * The main submission file migration is done in
+	 * PKPv3_3_0UpgradeMigration and that migration must
+	 * be run before this one.
+	 */
+	private function _migrateSubmissionFiles() {
+		Capsule::schema()->table('publication_galleys', function (Blueprint $table) {
+			$table->renameColumn('file_id', 'submission_file_id');
+			$table->foreign('submission_file_id')->references('submission_file_id')->on('submission_files');
```
[the remark this answers is not in the document]
Comment (NateWr) [reply classes/migration/upgrade/OPSv3_3_0UpgradeMigration.inc.php:166]: I have run the upgrades with these and had to adjust the order in which foreign keys were applied. And it seems to work with file deletions after moving the order of deleted objects around a bit. So I _think_ these are working ok but this is something that should get identified during testing.

**Pull request merged**